jupyter_console/interactiveshell.py

```diff
@@ -220,6 +231,43 @@ def handle_execute_reply(self, msg_id, timeout=None):
 
             self.execution_count = int(content["execution_count"] + 1)
 
+    def handle_is_complete_reply(self, msg_id, timeout=None):
+        """
+        Wait for a repsonse from the kernel, and return two values:
+            more? - (boolean) should the frontend ask for more input
+            indent - an indent string to prefix the input
+        Overloaded methods may want to examine the comeplete source. Its is
+        in the self._source_lines_buffered list.
+        """
+        ## Get the is_complete response:
+        msg = None
+        try:
+            msg = self.client.shell_channel.get_msg(block=True, timeout=timeout)
+        except Empty:
+            warn('The kernel did not respond to an is_complete_request. '
+                 'Setting `use_kernel_is_complete` to False.')
+            self.use_kernel_is_complete = False
+            return False, ""
+        ## Handle response:
+        if msg["parent_header"].get("msg_id", None) != msg_id:
+            warn('The kernel did not respond properly to an is_complete_request.')
```

Comment:
Can we show some details from the message it got - at least msg_type - to make it easier to debug if this does ever occur?